backends/bmv2/common/deparser.cpp

```diff
@@ -54,13 +55,15 @@ void DeparserConverter::convertDeparserBody(const IR::Vector<IR::StatOrDecl>* bo
                                 ref += "[" + Util::toString(i) + "]";
                                 result->append(ref);
                             }
-                        } else if (type->is<IR::Type_Header>()) {
+                        } else if (type->is<IR::Type_Header>() ||
+                                   type->is<IR::Type_HeaderUnion>()) {
```

Comment:
I'm not sure this is correct but maybe I'm missing something that i8s happening in the expression converter. The bmv2 JSON documentation says "For stacks and unions, all the header instances need to be listed in the appropriate order.". So shouldn't you iterate over all the union members here (the header instances that you are generating) and add each one to the `result` array?

Reply:
In fact stacks and header_unions were expanded into components by the expandEmit midend pass. So this code is useless.